### PR TITLE
Task/modify listing responses

### LINKED ISSUE #196: Modify the answer for the GET devices and GET groups operations
These operations currently return a direct array of the desired elements. This behavior should be changed to return an object with an attribute "count" indicating the number of results and an attribute "devices" or "services" with the actual result.

### LINKED ISSUE #199: List devices: the returned count value should be the total count of registered devices

### LINKED ISSUE #209: Asking for the list of configurations of a subservice return results for all services
If the subservice `/*` is used to retrieve all the configurations for a given service, all the subservices of all the services are retrieved instead. This behavior should be changed to ensure only subservices for the given service are retrieved.

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #209
Fixes #199
Fixes #196

### CODE REVIEW

File: test/unit/mongodb-registry-test.js [hunk 1]
```diff
@@ -339,4 +339,59 @@ describe('MongoDB Device Registry', function() {
...
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                async.map(devices, iotAgentLib.register, function(error, results) {
+                    done();
+                });
+            });
+        });
+        afterEach(function(done) {
+            iotAgentLib.clearRegistry(done);
+        });
+        it('should return the limited nunmber of devices', function(done) {
```
Comment (XavierVal) [inline test/unit/mongodb-registry-test.js]: typo "nunmber"

**Comment by AlvaroVega** [comment]
LGTM

File: test/unit/mongodb-registry-test.js (hunk 1 above)
Comment (dmoranj) [inline test/unit/mongodb-registry-test.js]: It's the number of nuns, really...

**Comment by anabelengp** [comment]
LGTM

**Comment by XavierVal** [comment]
LGTM

**Pull request merged**